### PR TITLE
Adding support for the extra parameters that can be used to authenticate via devise

### LINKED ISSUE #19: Token Authenticatable ignores "request_keys"
See http://www.rubydoc.info/github/plataformatec/devise/Devise/Models/Authenticatable for request_keys usage. You can basically ask it to pass extra data from the request (such as subdomain) so you can scope your lookups correctly.

The subdomain isn't passed in when using devise-token_authenticatable because of:

```ruby
        def find_for_token_authentication(conditions)
          find_for_authentication(authentication_token: conditions[Devise::TokenAuthenticatable.token_authentication_key])
        end
```

The conditions at this point contain the `subdomain`, but don't make their way in to `find_for_authentication` which in my case expects it.

### PR DESCRIPTION
**Pull request by brightbits**
Closes #19

### CODE REVIEW

File: lib/devise/token_authenticatable/model.rb [hunk 1]
```diff
@@ -32,7 +32,12 @@ module TokenAuthenticatable
       module ClassMethods
 
         def find_for_token_authentication(conditions)
-          find_for_authentication(authentication_token: conditions[Devise::TokenAuthenticatable.token_authentication_key])
+          conditions = conditions.dup
```
Comment (baschtl) [inline lib/devise/token_authenticatable/model.rb]: Please, don't overwrite a method argument. Just name the variable differently, e.g., `auth_conditions`.

File: spec/models/devise/token_authenticatable/model_spec.rb [hunk 2]
```diff
@@ -51,6 +51,16 @@
...
       end
 
+      it "should authenticate with all the options passed in, not just the auth_token" do
+        conditions = {facebook_token: entity.facebook_token, auth_token: entity.authentication_token}
+        expected_conditions = {facebook_token: entity.facebook_token, authentication_token: entity.authentication_token}
+
+        expect(described_class).to receive(:find_for_authentication).with(expected_conditions).and_call_original
+
+        authenticated_entity = described_class.find_for_token_authentication(conditions)
+        expect(entity.authentication_token).to eq(authenticated_entity.authentication_token)
```
Comment (baschtl) [inline spec/models/devise/token_authenticatable/model_spec.rb]: Please, only use one expectation per example as this makes debugging easier. Write another example for the second expectation.

**Comment by baschtl** [comment]
Beyond the mentioned things this is good to merge. Thanks a lot!

File: lib/devise/token_authenticatable/model.rb (hunk 1 above)
Comment (brightbits) [inline lib/devise/token_authenticatable/model.rb]: Changing

**Comment by brightbits** [comment]
Renamed the dupped parameter, removed the second expectation, it wasn't actually testing anything beyond that code outside the gems control did what I wanted it to, all the test cares about is that the extra parameters make it to the `find_for_authentication` method.

**Comment by baschtl** [comment]
@brightbits Thanks again. I will release a new version as soon as possible.

**Comment by brightbits** [comment]
Cheers :+1:

**Pull request merged**